### ISSUE TITLE
Add Expandable Combat Forces Breakdown to Combat Log

### ISSUE THREAD

**Issue by geoffthemedio** [freeorion/freeorion#1583]
In the combat log, it's a bit confusing at first to see ships mentioned as doing thing, without being first listed as participating in the battle. For the initial and lost forces numbers, a clickable expand button could be added that would show a list of all the relevant ships when expanded, similar to the expandable details about individual attacks in the log.
![combat_log_forces_breakdown](https://cloud.githubusercontent.com/assets/11386908/26398088/b9f06bf2-4078-11e7-9d07-4cd23900d0fa.png)

**Comment by mmoderau** [comment]
I have a *WIP* on this, but could really use a save game with interesting enough battle to test this out (combat with multiple factions, planets, ships, casualties, etc). Should I just go and ask on the forums about it?

I don't really have time to try and create such scenario myself.

**Comment by mmoderau** [comment]
Also, to make this look good, I would need to use the string formatting functions, that are in `CombatEvents.cpp` now.. and marked with a **TODO** to move them somewhere else.

Should I create a a separate MR for this before I create a MR for #1583?

**Comment by geoffthemedio** [comment]
This save as a hopefully interesting combat sitrep:
![combat_sitreps](https://user-images.githubusercontent.com/11386908/68850876-16e30980-06d5-11ea-8c14-3b27c012034a.png)
[combat_sitreps.zip](https://github.com/freeorion/freeorion/files/3846040/combat_sitreps.zip)

The save also has some ship design errors during loading, but hopefully that's not a problem.

**Comment by geoffthemedio** [comment]
I'd just implement your feature, then worry about code reorganization later.

**Comment by mmoderau** [comment]
> This save as a hopefully interesting combat sitrep:

Woah, that's precisely what I needed, thank you!

> I'd just implement your feature, then worry about code reorganization later.

This will mean literally copying and pasting code from one anonymous name space to another in different file.

I guess I can mark the copied code with a `//TODO` comment...

**Comment by mmoderau** [comment]
Here's my WIP using your savegame:
![obraz](https://user-images.githubusercontent.com/11045567/68891424-15d7c980-0721-11ea-85c3-1f347e225f6b.png)

*TODO:*
- Sort by object type (planets, ships) or divide into additional accordion subsections
- Sort by name
- Linkify

**Comment by mmoderau** [comment]
Also, I'll add another line break before "Lost:".

**Comment by mmoderau** [comment]
Should I use `FlexibleFormatList`? It is neat, but it reduces the maximum number of displayed elements to 10 and it requires a header string (even if I use an empty string for the header, there will be always one space between the "empty" header and the items).

I guess not.

**Comment by geoffthemedio** [comment]
> Should I use `FlexibleFormatList`

Not in this context.

**Comment by mmoderau** [comment]
This is funny. If I build the string by combining all the objects surrounded with proper tags, `LazyScrollerLinkText` *does not linkify the last element*, but if I append any non-link text to the string (tried with `"lol"`, assuming other values would work too), then the last entry is properly linkified.

**Comment by mmoderau** [comment]
Hm, adding a zero-width space does not fix the last element. experimenting with a space, and the will try a dot.

Fixing `LinkText` is outside of scope for this issue.

**Comment by geoffthemedio** [comment]
> Fixing `LinkText` is outside of scope for this issue.

Agreed

**Comment by mmoderau** [comment]
Added pull request: #2654.

**Comment by mmoderau** [comment]
Here's the same scenario but with my PR:
![obraz](https://user-images.githubusercontent.com/11045567/69485348-65fc0d80-0e3e-11ea-94b3-447008e1abec.png)

**Comment by mmoderau** [comment]
In the pull request, @geoffthemedio noted that the current criteria for displaying "forces" in combat don't make much sense.

Currently it's one of:

1. Is a ship
1. Has population meter with positive value

Should I add a separate issue to clean this up, or should this be the next iteration implementation for this issue?

**Comment by Vezzra** [comment]
~~With #2654 merged, can we close this as resolved?~~

EDIT: Sorry, never mind, didn't realize the feature apparently isn't completely implemented yet.

**Comment by mmoderau** [comment]
@Vezzra, it's actually debatable, if the feature is implemented completely or not, hence my question above.

The changes as stated in original description were provided. During implementation it came to light, that the set of objects counted as forces does not necessarily make sense - but here I think some discussion is required. Should planets appear in "lost" section, for example? Under what circumstances? Etc.

The changes I made enumerates the objects that were counted and groups them by criterion used - I did not change the criteria themselves.
